### PR TITLE
[feat 4898]: Dashboard filter changes

### PR DESCRIPTION
**Pull request by OscarBakker**
**context**
implementing some requested changes of ux, plus making the opening menus mobile firstisch.

Ticket: [SIG-4898](https://gemeente-amsterdam.atlassian.net/browse/SIG-4898)

## Signalen

Before opening a pull request, please ensure:

- Make sure your PR title follows naming conventions: [feat-1234]: name feature
- Double-check your branch is based on `main` and targets `main`
- Pull request has tests (we are going for 100% coverage!)
- Code is well-commented, linted and follows project conventions
- Committed source code is headed by the correct SPDX license expression

Be kind to code reviewers, please try to keep pull requests as small and focused as possible :)

[SIG-4898]: https://datapunt.atlassian.net/browse/SIG-4898?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/signals/incident-management/containers/Dashboard/components/Filter/styled.tsx [hunk 1]
```diff
@@ -61,41 +70,51 @@ export const RefreshIcon = styled(Refresh)`
...
+  optionsOffsetLeft?: number
   optionsTotal: number
 }>`
   display: grid;
   position: absolute;
-  width: ${themeSpacing(175)};
   line-height: 2rem;
   padding: 0;
   margin: 0;
+  width: 100%;
```
Comment (OscarBakker) [inline src/signals/incident-management/containers/Dashboard/components/Filter/styled.tsx:107]: on mobile show max width 100

File: src/signals/incident-management/containers/Dashboard/components/Filter/styled.tsx [hunk 2]
```diff
@@ -28,9 +32,14 @@ export const Select = styled.div<{
...
   margin-right: ${themeSpacing(9)};
+  height: ${themeSpacing(14)};
+  margin-left: ${themeSpacing(12)};
 
-  &:first-of-type {
-    margin-left: ${themeSpacing(12)};
+  @media ${breakpoint('min-width', 'tabletM')} {
+    margin-left: unset;
+    &:first-of-type {
+      margin-left: ${themeSpacing(12)};
```
Comment (vdegraaf) [inline src/signals/incident-management/containers/Dashboard/components/Filter/styled.tsx:41]: According to desgin the left-margin should be 61px.

**Review by vdegraaf — COMMENTED** [review]
In the feedback it says only show categories for which the department is responsible. Currently all categories to which the department has access are shown.

File: src/signals/incident-management/containers/Dashboard/components/Filter/styled.tsx (hunk 2 above)
Comment (OscarBakker) [reply src/signals/incident-management/containers/Dashboard/components/Filter/styled.tsx:41]: will do!

**Comment by OscarBakker** [comment]
> In the feedback it says only show categories for which the department is responsible. Currently all categories to which the department has access are shown.

wondering how to select, will ask BE

File: src/signals/incident-management/containers/Dashboard/components/Filter/SelectList.tsx [hunk 3]
```diff
@@ -129,14 +133,18 @@ const SelectList = ({ filterActiveName, setFilterActiveName }: Props) => {
...
-      <OptionListDropdown active={!!activeFilter?.name}>
-        {activeFilter?.name && isNumber(optionsOffsetLeftRef.current) && (
-          <OptionsList
-            optionsOffsetLeft={optionsOffsetLeftRef.current}
-            activeFilter={activeFilter}
-            setFilterNameActive={setFilterActiveName}
-          />
-        )}
+      <OptionListDropdown
+        active={!!activeFilter?.name}
```
Comment (SireeKoolenWijkstra) [inline src/signals/incident-management/containers/Dashboard/components/Filter/SelectList.tsx:137]: Double exclamation remarks? Why not just no remarks then (a quick google search didn't say anything about double exclamation remarks)?

File: src/signals/incident-management/containers/Dashboard/components/Filter/styled.tsx (hunk 2 above)
Comment (SireeKoolenWijkstra) [reply src/signals/incident-management/containers/Dashboard/components/Filter/styled.tsx:41]: Ah, friday Linda told me to keep a margin of 40. For the components at least. Not sure if that also applies to the filter.

**Review by SireeKoolenWijkstra — CHANGES_REQUESTED** [review]

File: src/signals/incident-management/containers/Dashboard/components/Filter/SelectList.tsx (hunk 3 above)
Comment (vdegraaf) [reply src/signals/incident-management/containers/Dashboard/components/Filter/SelectList.tsx:137]: You should really fix your google. You're missing out!

Screenshot 2023-02-07 at 12 01 39

Comment (OscarBakker) [reply src/signals/incident-management/containers/Dashboard/components/Filter/SelectList.tsx:137]: 🥷

**Review by SireeKoolenWijkstra — APPROVED** [review]

File: src/signals/incident-management/containers/Dashboard/components/Filter/SelectList.tsx (hunk 3 above)
Comment (SireeKoolenWijkstra) [reply src/signals/incident-management/containers/Dashboard/components/Filter/SelectList.tsx:137]: I would if I knew how.

**Comment by OscarBakker** [comment]
todo 
- [ ] build test for hook

File: src/signals/incident-management/containers/Dashboard/hooks/useDepartments.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+import CONFIGURATION from 'shared/services/configuration/configuration'
+import type { ApplicationRootState } from 'types'
+import type { Department } from 'types/api/incident'
+
+const cachedDepartments: { [key: string]: any } = {}
+export const useDepartments = (): {
+  departments: Department[]
+  isLoading: boolean
+} => {
+  const [departments, setDepartments] = useState<any>()
```
Comment (vdegraaf) [inline src/signals/incident-management/containers/Dashboard/hooks/useDepartments.ts:18]: Can have a proper type i think.

**Review by vdegraaf — COMMENTED** [review]
Nice, some comments:
- The whole background of the dashboard is grey (should be white as it was)
- Department Politie, for example, is not responsible for any categories. Should they be in the list?

**Comment by OscarBakker** [comment]
> Nice, some comments:
> 
> * The whole background of the dashboard is grey (should be white as it was)

- [ ] fix
> * Department Politie, for example, is not responsible for any categories. Should they be in the list?
yeah I was guessing to leave them out, though I'll check it with ux first! - [ ] check with ux

**Review by SireeKoolenWijkstra — COMMENTED** [review]
Hi Oscar,

- [ ] Category doesn't seem to function properly anymore. When I select department Port of Amsterdam, or Marktbureau, or Politie, or Stadsdelen, or Omgevingsdienst, or V&OR VRI no categories are available. If it is because there are no categories, then I would expect not to be able to click on Categories or something like that. Now I think they are missing.
- [ ] There are a couple of departments with only a view (or just one) category, for example FB. But when you click on category the whole screen is filled with grey I would expect only a limited amount of grey, just enough to encompass the text but no more.

**Comment by OscarBakker** [comment]
> Hi Oscar,
> 
> * [ ]  Category doesn't seem to function properly anymore. When I select department Port of Amsterdam, or Marktbureau, or Politie, or Stadsdelen, or Omgevingsdienst, or V&OR VRI no categories are available. If it is because there are no categories, then I would expect not to be able to click on Categories or something like that. Now I think they are missing.

I don't think you got the latest version, thats just fixed

> * [ ]  There are a couple of departments with only a view (or just one) category, for example FB. But when you click on category the whole screen is filled with grey I would expect only a limited amount of grey, just enough to encompass the text but no more.

to the second, thats not according ux. We could have a limited dropdown, but we chose a full one

File: src/signals/incident-management/containers/Dashboard/components/Filter/Filter.test.tsx [hunk 5]
```diff
@@ -78,7 +95,7 @@ describe('FilterComponent', () => {
   })
 
   it('should select a department and thus change selectable categories', async () => {
```
Comment (vdegraaf) [inline src/signals/incident-management/containers/Dashboard/components/Filter/Filter.test.tsx:97]: not an async function

File: src/signals/incident-management/containers/Dashboard/components/Filter/Filter.test.tsx [hunk 6]
```diff
@@ -191,7 +208,7 @@ describe('FilterComponent', () => {
   })
 
   it('should select a department, a custom category and reset back and call callback each time', async () => {
```
Comment (vdegraaf) [inline src/signals/incident-management/containers/Dashboard/components/Filter/Filter.test.tsx:210]: not an async function

File: src/signals/incident-management/containers/Dashboard/components/Filter/Filter.test.tsx [hunk 7]
```diff
@@ -444,4 +462,24 @@ describe('FilterComponent', () => {
 
     expect(mockSetDashboardFilter.mock.calls[1]).toEqual([{}])
   })
+
+  it('should show a spinner', async () => {
```
Comment (vdegraaf) [inline src/signals/incident-management/containers/Dashboard/components/Filter/Filter.test.tsx:466]: not an async function

File: src/signals/incident-management/containers/Dashboard/hooks/useDepartments.test.ts [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+
+    jest.spyOn(reactRedux, 'useSelector').mockReturnValue({ list: departments })
+    jest.spyOn(React, 'useContext').mockReturnValue({
+      departmentsWithResponsibleCategories: {
+        departments,
+      },
+    })
+  })
+
+  it('should return departments', async () => {
```
Comment (vdegraaf) [inline src/signals/incident-management/containers/Dashboard/hooks/useDepartments.test.ts:51]: not an async function

File: src/signals/incident-management/containers/Dashboard/hooks/useDepartments.test.ts [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+          value: 'CCA',
+          display: 'CCA',
+          category_names: ['A'],
+        },
+      ],
+      isLoading: false,
+    })
+  })
+
+  it('should return undefined departments when data from useFetchAll is undefined', async () => {
```
Comment (vdegraaf) [inline src/signals/incident-management/containers/Dashboard/hooks/useDepartments.test.ts:72]: not an async function

File: src/signals/incident-management/containers/Dashboard/hooks/useDepartments.test.ts [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+
+    const { result } = renderHook(useDepartments)
+
+    expect(result.current).toEqual({
+      departments: undefined,
+      isLoading: false,
+    })
+  })
+
+  it('should return undefined departments when departments from redux are empty', async () => {
```
Comment (vdegraaf) [inline src/signals/incident-management/containers/Dashboard/hooks/useDepartments.test.ts:87]: not an async function

**Review by vdegraaf — APPROVED** [review]
Looking good! Approved! 
Do have two comments:
- you can remove all `async` in the tests, these are not async functions
- it is very hard to look for a category since they are in random order. Is it an idea to alphabetize them?

**Comment by OscarBakker** [comment]
> Looking good! Approved! Do have two comments:
> 
> * you can remove all `async` in the tests, these are not async functions

strange, will remove!

> * it is very hard to look for a category since they are in random order. Is it an idea to alphabetize them?

yeah why not, no obvious other way of ordering

**Pull request merged**